src/awkward/contents/unionarray.py

```diff
@@ -1293,7 +1292,15 @@ def numbers_to_type(self, name):
         )
 
     def _is_unique(self, negaxis, starts, parents, outlength):
-        simplified = self.simplify_uniontype(True, True)
+        simplified = type(self).simplified(
+            self._tags,
+            self._index,
+            self._contents,
+            parameters=self._parameters,
+            backend=self._backend,
+            merge=True,
+            mergebool=True,
+        )
```

Comment:
This one re-simplifies the inputs because the merging strategy is more aggressive than normal (`mergebool=True`), so this is to prevent any changes in behavior.